lib/SILGen/SILGenThunk.cpp

```diff
@@ -159,10 +161,32 @@ SILGenModule::getOrCreateForeignAsyncCompletionHandlerImplFunction(
   std::copy(blockType->getParameters().begin(),
             blockType->getParameters().end(),
             std::back_inserter(implArgs));
+
+  const clang::Type *clangType = blockType->getClangTypeInfo().getType();
+  const clang::Type *newClangType = nullptr;
+  if (auto blockPtrTy = dyn_cast_or_null<clang::BlockPointerType>(clangType)) {
```

Comment:
Can this be null?  Can the cast fail?

Reply:
It can be null because `UseClangFunctionTypes` might be turned off (and this fix is independent of that). It can't fail though.

Reply:
Fixed.